internal/pkg/utils/testhelper/runner/test.go

```diff
@@ -460,27 +475,61 @@ func (t *Test) runRequests(apiURL string, requestDecoratorFn func(*APIRequest))
 		// Decorate the request
 		requestDecoratorFn(request)
 
-		// Send request
-		r.SetContext(context.WithValue(r.Context(), dumpDirCtxKey, requestDir))
-		resp, err := r.Execute(request.Method, request.Path)
-		assert.NoError(t.t, err)
-
-		// Dump raw HTTP response
-		if err == nil {
-			respDump, err := httputil.DumpResponse(resp.RawResponse, false)
-			assert.NoError(t.t, err)
-			assert.NoError(t.t, t.workingDirFS.WriteFile(filesystem.NewRawFile(filesystem.Join(requestDir, "response.txt"), string(respDump)+string(resp.Body()))))
+		// Find and replace references to other requests in the request path
+		reqPath, err := processPathReference(request.Path, requests)
+		if err != nil {
+			t.t.Fatal(err)
 		}
 
-		// Compare response body
-		expectedRespBody := t.ReadFileFromTestDir(filesystem.Join(requestDir, "expected-response.json"))
+		// Set repeat requests timeout
+		reqTimeout := 60 * time.Second
+		if request.Repeat.Timeout > 0 {
+			reqTimeout = time.Duration(request.Repeat.Timeout) * time.Second
+		}
 
-		// Decode && encode json to unite indentation of the response with expected-response.json
+		var resp *resty.Response
 		respMap := orderedmap.New()
-		if resp.String() != "" {
-			err = json.DecodeString(resp.String(), &respMap)
+		// Allow repeating the request until a condition is met
+		for start := time.Now(); time.Since(start) < reqTimeout; {
+			// Send request
+			r.SetContext(context.WithValue(r.Context(), dumpDirCtxKey, requestDir))
+			resp, err = r.Execute(request.Method, reqPath)
+			assert.NoError(t.t, err)
+
+			// Dump raw HTTP response
+			if err == nil {
+				respDump, err := httputil.DumpResponse(resp.RawResponse, false)
+				assert.NoError(t.t, err)
+				assert.NoError(t.t, t.workingDirFS.WriteFile(filesystem.NewRawFile(filesystem.Join(requestDir, "response.txt"), string(respDump)+string(resp.Body()))))
+			}
+
+			// Get the response body
+			// Decode && encode json to unite indentation of the response with expected-response.json
+			if resp.String() != "" {
+				err = json.DecodeString(resp.String(), &respMap)
+			}
+			requests[requestDir].Response = respMap
+			assert.NoError(t.t, err)
+
+			// Run only once if there is no repeat until condition
+			if request.Repeat.Until == "" {
+				break
+			}
+
+			// Evaluate repeat until condition
+			repeatUntilExp, err := goValuate.NewEvaluableExpression(request.Repeat.Until)
+			if err != nil {
+				t.t.Fatal(errors.Errorf("cannot compile repeat until expression: %w", err))
+			}
+			repeatUntilVal, err := repeatUntilExp.Evaluate(respMap.ToMap())
+			if err != nil {
+				t.t.Fatal(errors.Errorf("cannot evaluate repeat until expression: %w", err))
+			}
+			if repeatUntilVal.(bool) {
+				break
+			}
```

Comment:
omg, tady jsem zapomněl ještě nějaký sleep. 🙂

Reply:
Nakonec jsem přidal možnost parametrizovat i ten sleep:

https://github.com/keboola/keboola-as-code/blob/d85ddbec41605c7fc9e13c9d499869ee8ac72d8c/docs/e2etests.md#L183-L200

Když už se dá definovat timeout, tak by to bez něj nedávalo smysl. Imho to stačí takto fixně, pro testy asi není třeba vymýšlet nějaký sofistikovaný backoffy.